src/datasets/summarizations/trend-weekly-comparison-average.summarization.ts

```diff
@@ -0,0 +1,41 @@
+
+import { Summary } from './types';
+import { TimeSeriesPoint } from '../metas/types';
+import { cacheSummaries } from './utils/commons';
+import {
+  groupPointsByXWeek,
+} from './utils/time-series';
+import { formatY } from '../../utils/formatters';
+
+export function queryFactory(points: TimeSeriesPoint[]) {
+  return cacheSummaries(() => {
+    // Only consider weeks with more than 3 days when creating summaries
+    // Weeks with 3 days or less are considered to belong to last/next 30 days
+    const weekPointArrays = groupPointsByXWeek(points).filter(weekPoints => weekPoints.length >= 4);
+    const nWeeks = weekPointArrays.length;
+
+    const weekAverages = weekPointArrays.map(weekPoints => weekPoints.reduce((p, { y }) => p + y, 0) / weekPoints.length);

```

Comment:
"weeklyAverages" I think you are referring to multiple weeks here.

Reply:
I prefer to keep the prefix `week` to align with other variables like `weekPoints, but I just renamed it to `weekYAverages` to improve readability.